cmd/tfchainc/erc20cmd.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/spf13/cobra"
+	"github.com/threefoldfoundation/tfchain/cmd/tfchainc/internal"
+	"github.com/threefoldtech/rivine/pkg/cli"
+)
+
+// createERC20Cmd creates rootcommand for ERC20 and adds a subcommand
+// if rootcommand executed the user will also see the output of the syncing status of ethereum
+func createERC20Cmd(client *internal.CommandLineClient) *cobra.Command {
+	erc20SubCmds := &erc20SubCmds{cli: client}
+
+	// define Rootcommand
+	var (
+		rootCmd = &cobra.Command{
+			Use:   "erc20",
+			Short: "Perform erc20 actions",
+			Long:  "Perform erc20 actions",
+			Run:   erc20SubCmds.getSyncingStatus,
+		}
+		getSyncingStatusCmd = &cobra.Command{
+			Use:   "syncstatus",
+			Short: "Get the ethereum sync status",
+			Long:  `Get the ethereum chain sync status.`,
+			Run:   erc20SubCmds.getSyncingStatus,
+		}
+	)
+
+	rootCmd.AddCommand(getSyncingStatusCmd)
+
+	// register flags
+	getSyncingStatusCmd.Flags().Var(
+		cli.NewEncodingTypeFlag(0, &erc20SubCmds.getSyncingStatusCfg.EncodingType, 0), "encoding",
+		cli.EncodingTypeFlagDescription(0))
+
+	return rootCmd
+}
+
+type erc20SubCmds struct {
+	cli                 *internal.CommandLineClient
+	getSyncingStatusCfg struct {
+		EncodingType cli.EncodingType
+	}
+}
+
+func (erc20SubCmds *erc20SubCmds) getSyncingStatus(cmd *cobra.Command, args []string) {
+	erc20cmds := internal.NewERC20Client(erc20SubCmds.cli)
+
+	syncingStatus, err := erc20cmds.GetSyncingStatus()
+	if err != nil {
+		cli.DieWithError("error while fetching the syncing status", err)
+	}
+
+	// encode depending on the encoding flag
+	var encode func(interface{}) error
+	switch erc20SubCmds.getSyncingStatusCfg.EncodingType {
+	case cli.EncodingTypeHuman:
```

Comment:
You go way too complex here. I would do the encoding something as follows:

```go
switch erc20SubCmds.getSyncingStatusCfg.EncodingType {
case cli.EncodingTypeHuman:
     if not-syncing {
          fmt.Println("ERC20 node is not synchronizing") // TODO: we might want to still be able to print current height somehow
     } else {
          fmt.Printf(`ERC20 node is currently synchronizing...
Starting Height: %d
Current Height: %d,
blabla: %d
`, val, val, val)
      }
case cli.EncodingTypeJSON:
    err := json.NewEncoder(os.Stdout).Encode(syncingStatus)
    if err != nil {
          cli.DieWithError("failed to encode syncing status", err)
     }
}
```

Reply:
Changed in: https://github.com/threefoldfoundation/tfchain/pull/307/commits/dc958a46503e7ab519c9d29c5d94598f09ea7d5c